filters/f_hwtransfer.c

```diff
@@ -291,8 +291,36 @@ static bool probe_formats(struct mp_filter *f, int hw_imgfmt, bool use_conversio
         if (!cur->av_device_ref)
             continue;
         cstr = av_hwdevice_get_hwframe_constraints(cur->av_device_ref, NULL);
-        if (!cstr)
-            continue;
+        if (!cstr) {
+            /*
+             * Some hwcontexts do not implement constraints, and so cannot
+             * report supported formats, so cobble something together from our
+             * static metadata.
+             */
+            cstr = av_malloc(sizeof(AVHWFramesConstraints));
```

Comment:
probably not really safe to let `av_hwframe_constraints_free` free this
how about using talloc here?

Reply:
I wanted to avoid having a separate cleanup path. The ffmpeg code uses `av_free`, which theoretically could change, but is a safe behaviour to assume in practice.